### PR TITLE
Delete legacy blob component

### PR DESCRIPTION
**Pull request by taras-yemets**
Removes legacy (table-based) blob component. 
CodeMirror has been a default blob viewer since Sourcegraph `4.5.0`. Table-based legacy blob viewer was [left as an opt-out option](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@5380d0c887e484a15c3c96783f3b49187687addf/-/blob/CHANGELOG.md?L55&subtree=true&view=code) in case we encounter major issues with the CodeMirror blob viewer. Luckily no significant issues were reported so we can remove the legacy (table-based) blob view. 

## Test plan
- CI passes

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.75% (-20.67 kb) 🔽 | -0.12% (-17.56 kb) 🔽 | 0.03% (+3.11 kb) | -0.53% (-4) 🔽 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/taras-yemets/delete-legacy-blob/compare-report.html;tab=live_object) for a full comparison between the commits 586ca92 and bdda34a or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by philipp-spiess** [comment]
Awesome! Does this mean we can also get rid of the fact that the first of the two `Blob` queries includes the legacy HTML rendered (IIRC this was necessary for backward compatibility to the old renderer).

For this first query we only need the `content` as raw text, the HTML highlight is just discarded IIRC. 

Screenshot 2023-04-26 at 14 58 50

**Comment by taras-yemets** [comment]
>Does this mean we can also get rid of the fact that the first of the two Blob queries includes the legacy HTML rendered (IIRC this was necessary for backward compatibility to the old renderer).
For this first query we only need the content as raw text, the HTML highlight is just discarded IIRC.

I guess we can.

Screenshot 2023-04-26 at 16 43 12

Setting the default value `HTMLResponseFormat` to `JSON_SCIP`(https://github.com/sourcegraph/sourcegraph/commit/a8efd30295e764064a30ce9842e5601e4ec6c84d) eliminates the first Blob query. 
Now [both `fetchBlob` calls](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@e31bd4d938601641401fe211828bd6b2d5b0f58d/-/blob/client/web/src/repo/blob/BlobPage.tsx?L292) use _almost_ the same arguments and [are memoized](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@e31bd4d/-/blob/client/web/src/repo/blob/backend.ts?L32-38). When I say _almost_ I mean that `highlightedBlobInfoOrError` may resolve even faster as it [may have `disableTimeout` param set to `true`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@e31bd4d938601641401fe211828bd6b2d5b0f58d/-/blob/client/web/src/repo/blob/BlobPage.tsx?L257). In this case, do we even need `formattedBlobInfoOrError` (cc: @umpox)?

**Comment by taras-yemets** [comment]
We can even remove the [`format`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@e31bd4d938601641401fe211828bd6b2d5b0f58d/-/blob/client/web/src/repo/blob/backend.ts?L45) field from `FetchBlobOptions` and [pass `HighlightResponseFormat.JSON_SCIP` by default](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@e31bd4d938601641401fe211828bd6b2d5b0f58d/-/blob/client/web/src/repo/blob/backend.ts?L110). WDYT, @philipp-spiess?

**Comment by umpox** [comment]
I think we will still see code load noticeably faster with `formattedBlobInfoOrError` (and `format: HighlightResponseFormat.HTML_PLAINTEXT` set) mainly for medium/large/very large files.

I did a quick test loading [this file](https://sourcegraph.test:3443/github.com/sourcegraph/sourcegraph/-/blob/enterprise/internal/database/mocks_temp.go):
- Time to see code without plaintext: `2.1s`
- Time to see code with plaintext: `567ms`


That files pretty big but I still see a 50ms-100ms improvement when loading medium sized files. For small files the difference is barely noticeable.


I think we should keep it. Btw, `disableTimeout` isn't actually used (in the backend) for the plaintext query IIRC, it's only for highlighted code.

File: client/web/src/repo/blob/backend.ts [hunk 1]
```diff
@@ -66,11 +65,6 @@ export const fetchBlob = memoizeObservable(
             visibleIndexID,
         } = applyDefaultValuesToFetchBlobOptions(options)
 
-        // We only want to include HTML data if explicitly requested. We always
-        // include LSIF because this is used for languages that are configured
-        // to be processed with tree sitter (and is used when explicitly
-        // requested via JSON_SCIP).
-        const html = [HighlightResponseFormat.HTML_PLAINTEXT, HighlightResponseFormat.HTML_HIGHLIGHT].includes(format)
```
Comment (umpox) [inline client/web/src/repo/blob/backend.ts:73]: I think we'll get a perf boost from this PR. Looks like we were fetching `html` for plaintext responses when we didn't actually need to, nice!

**Review by umpox — APPROVED** [review]
Looks good! Exciting!

**Pull request merged**